tools/clean.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python3
+"""
+Clean the world.
+
+Note that this is probably only a partial clean - in particular, no attempt
+is made to clean the iOS/Swift world, because markh doesn't know what that
+involves.
+
+Please make a PR with anything you notice that should be cleaned but isn't!
+"""
+
+import argparse
+import subprocess
+from pathlib import Path
+import shlex
+import shutil
+
+PROJECT_ROOT = Path(__file__).parent.parent
+
+def run_command(dry_run, cmdline, **kwargs):
+    print('Executing:', ' '.join(shlex.quote(str(part)) for part in cmdline))
+    if not dry_run:
+        subprocess.check_call(cmdline, **kwargs)
+
+def find_generated_directories(look_dir):
+    for child in look_dir.iterdir():
+        if child.name == 'support':
+            for sub in find_generated_directories(child):
+                yield sub
+        else:
+            # `android/build` directories should be removed.
+            sub = child / 'android' / 'build'
+            if sub.is_dir():
+                yield sub
+            # TODO: ios/swift?
+
+def clean_mobile(dry_run):
+    for to_rm in find_generated_directories(PROJECT_ROOT / "components"):
+        print('Removing:', to_rm)
+        if not dry_run:
+            shutil.rmtree(to_rm)
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('-n', '--dry-run', dest="dry_run", action="store_true",
+                        help='show what would be executed/removed without actually doing it.')
+    return parser.parse_args()
+
+def main():
+    args = parse_args()
+    run_command(args.dry_run, ["cargo", "clean"])
```

Comment:
You mentioned in a comment about `./gradlew clean` and it seems worth adding here. In fact if there's android-specific stuff that *doesn't* get cleaned by `./gradlew clean`, it may be worth our time to try to fix that command to do it rather than working around it here.  (Or it may not, gradle's always an adventure...)